### PR TITLE
Fix bounding box center overflow for narrow dtypes

### LINKED ISSUE #9594: XYXY to CXCYWH conversion overflows for narrow dtypes
### Describe the bug

`torchvision.transforms.v2.functional.convert_bounding_box_format` can return an incorrect center when converting valid `XYXY` boxes to `CXCYWH` with a narrow dtype.

The internal `_xyxy_to_cxcywh` implementation first computes the width, but then evaluates the equivalent expression `(x1 * 2 + width) / 2`. The intermediate `x1 * 2` can overflow even when the input coordinates, width, and mathematically correct center are all representable in the input dtype.

For `uint8`, the center wraps from 210 to 82. For `float16`, the center becomes infinity instead of 40960.

```python
import torch
from torchvision import tv_tensors
from torchvision.transforms.v2 import functional as F

cases = [
    (torch.uint8, [200, 10, 220, 30]),
    (torch.float16, [32768, 0, 49152, 32]),
]

for dtype, box in cases:
    output = F.convert_bounding_box_format(
        torch.tensor([box], dtype=dtype),
        old_format=tv_tensors.BoundingBoxFormat.XYXY,
        new_format=tv_tensors.BoundingBoxFormat.CXCYWH,
    )
    print(dtype, output)
```

Actual output on current `main` at `34572106`:

```text
torch.uint8 tensor([[82, 20, 20, 20]], dtype=torch.uint8)
torch.float16 tensor([[   inf, 1.6000e+01, 1.6384e+04, 3.2000e+01]], dtype=torch.float16)
```

Expected output:

```text
torch.uint8 tensor([[210, 20, 20, 20]], dtype=torch.uint8)
torch.float16 tensor([[40960, 16, 16384, 32]], dtype=torch.float16)
```

The overflow can be avoided by evaluating the equivalent but safer expression `x1 + width / 2`. I would like to contribute that focused change with regression coverage for both in-place and out-of-place conversions.

I searched open and closed issues and pull requests for bounding-box center, `uint8`, and `float16` conversion overflow and did not find an existing report for this path.

AI disclosure: I used Codex to help audit the conversion arithmetic, search for duplicates, reproduce the failure, test the proposed expression, and draft this report. I reviewed the diagnosis, reproduction, and proposed scope and will personally handle follow-up.

### Versions

```text
TorchVision: current main at 34572106ad1f0ea95793e379751f8bb0cfeeac1c
PyTorch: 2.13.0+cpu
Python: 3.10.11 (64-bit)
OS: Microsoft Windows 11 Home Single Language 10.0.26200
CUDA available: False
NumPy: 2.2.6
```

### PR DESCRIPTION
**Pull request by aswanth-07**
## Summary

- avoid the overflowing `x1 * 2` intermediate in `XYXY` to `CXCYWH` conversion
- compute the equivalent center as `x1 + width / 2`
- add regression coverage for `uint8` and `float16`, in both in-place and out-of-place conversion

## Why

`_xyxy_to_cxcywh` first stores `x2 - x1` as the width, then computes the center through `(x1 * 2 + width) / 2`. For narrow dtypes, `x1 * 2` can overflow even when the input coordinates, width, and correct center all fit in the dtype.

For example, a valid `uint8` box with x coordinates 200 and 220 returns center 82 instead of 210, while a valid `float16` box with x coordinates 32768 and 49152 returns infinity instead of 40960. Reordering the same expression to `x1 + width / 2` avoids the unnecessary intermediate and retains integer floor rounding.

Fixes #9594.

## Validation

- full `TestConvertBoundingBoxFormat`: 184 passed, 30 expected failures
- 5,000 randomized valid boxes across uint8, int8, int16, int32, and int64
- float autograd smoke test
- `ufmt check` on both changed files
- changed lines introduce no new flake8 or mypy diagnostics; broader commands report only unchanged baseline findings
- `git diff --check`

The affected conversion is pure Python. The local source checkout used the compatible installed TorchVision extension and supplied only the newly introduced `qnms` schema required during import.

AI disclosure: I used Codex to help audit the conversion arithmetic, search for duplicates, implement and test the focused patch, and draft this PR. I reviewed and understand the change and will personally handle maintainer follow-up.

### CODE REVIEW

**Comment by aswanth-07** [comment]
Closing this PR after audit. Torchvision's CONTRIBUTING.md requires bug fixes to target an open help-wanted issue and comment on it first. Issue #9594 has no help-wanted label or assignment comment, so this PR does not meet the repository's stated policy. The source branch remains available.
